spec/services/preserved_object_handler_spec.rb

```diff
@@ -335,4 +335,42 @@
 
     it_behaves_like 'PreservedCopy does not exist', :confirm_version
   end
+
+  describe '#with_active_record_transaction_and_rescue' do
```

Comment:
should there be a test for each of:

- create
- create_after_validation
- update_version
- update_version_after_validation (doesn't exist yet)
- confirm_version

proving that they roll back when they are supposed to?  Or maybe proving that they use the method?

Reply:
i would actually like to do that, though i was having a bit of trouble coming up with a test for `update_version` that didn't feel too contrived.  but certainly possible.  at worst, we could force the pres obj update call to blow up, and confirm that the pres copy changes got rolled back.

i was also thinking it might be nice to test the private method specifically/explicitly, but maybe that's overkill?  see #324.

if we did add tests for the methods you suggested, would you prefer to do that as part of this PR, or as part of a follow on ticket (e.g. slightly change or expand the scope of #324)?

Reply:
I think you could either expand the scope of #324 or make a new ticket.

Reply:
expanded scope of #324